aea/cli/launch.py

```diff
@@ -121,7 +122,11 @@ def _launch_threads(click_context: click.Context, agents: List[Path]):
 def launch(click_context, agents: List[str], multithreaded: bool):
     """Launch many agents at the same time."""
     agents_directories = list(map(Path, list(OrderedDict.fromkeys(agents))))
-    if not multithreaded:
-        _launch_subprocesses(click_context, agents_directories)
+    if multithreaded:
+        failed = _launch_threads(click_context, agents_directories)
     else:
-        _launch_threads(click_context, agents_directories)
+        failed = _launch_subprocesses(click_context, agents_directories)
+    logger.debug(f"Exit cli. code: {failed}")
+    sys.stdout.flush()
+    sys.stderr.flush()
+    sys.exit(failed)
```

Comment:
Why is the flush needed? Just curious...

Reply:
I suppose there was a problem with process stdout flush on close, so pexpect can not get final lines. but solved. flush removed